packages/main/src/SliderBase.js

```diff
@@ -0,0 +1,623 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import Float from "@ui5/webcomponents-base/dist/types/Float.js";
+import Integer from "@ui5/webcomponents-base/dist/types/Integer.js";
+import ResizeHandler from "@ui5/webcomponents-base/dist/delegate/ResizeHandler.js";
+import { getTheme } from "@ui5/webcomponents-base/dist/config/Theme.js";
+
+// Styles
+import styles from "./generated/themes/SliderBase.css.js";
+
+/**
+ * @public
+ */
+const metadata = {
+	properties: /** @lends sap.ui.webcomponents.main.SliderBase.prototype */  {
+		/**
+		 * Defines the minimum value of the slider
+		 * <br><br>
+		 *
+		 * @type {Float}
+		 * @defaultvalue 0
+		 * @public
+		 */
+		min: {
+			type: Float,
+			defaultValue: 0,
+		},
+		/**
+		 * Defines the maximum value of the slide
+		 * <br><br>
+		 *
+		 * @type {Float}
+		 * @defaultvalue 100
+		 * @public
+		 */
+		max: {
+			type: Float,
+			defaultValue: 100,
+		},
+		/**
+		 * Defines the size of the slider's selection intervals. (e.g. min = 0, max = 10, step = 5 would result in possible selection of the values 0, 5, 10).
+		 * If set to 0 the slider handle movement is disabled. When negative number or value other than a number, the component fallbacks to its default value.
+		 * <br><br>
+		 *
+		 * @type {Integer}
+		 * @defaultvalue 1
+		 * @public
+		 */
+		step: {
+			type: Float,
+			defaultValue: 1,
+		},
+		/**
+		 * Displays a label with a value on every N-th step. The step and tickmarks properties must be enabled.
+		 * Example - if the step value is set to 2 and the label interval is also specified to 2 - than every second
+		 * tickmark will be labelled, which means every 4th value number.
+		 *
+		 * @type {Integer}
+		 * @defaultvalue 1
+		 * @public
+		 */
+		labelInterval: {
+			type: Integer,
+			defaultValue: 0,
+		},
+		/**
+		 * Enables tick marks visualization for each step. The step value must not be set to 0.
+		 * <br><br>
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		 tickmarks: {
+			type: Boolean,
+		},
+		/**
+		 * Enables handle tooltip displaying the current value.
+		 * <br><br>
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		showTooltip: {
+			type: Boolean,
+		},
+		/**
+		 * Defines whether the <code>ui5-slider</code> is in disabled state.
```

Comment:
I think it is better to use `<code>slider</code>` instead of `<code>ui5-slider</code>` in the metadata of SliderBase, because when we create the RangeSlider, it will appear in its API reference and  `<code>ui5-slider</code>` would be wrong as the RangeSlider's tag would be ui5-range-slider. So, check the metadata